cmd/podman/farm/build.go

```diff
@@ -0,0 +1,129 @@
+package farm
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/cmd/podman/utils"
+	"github.com/containers/podman/v4/pkg/farm"
+	"github.com/containers/storage"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type buildOptions struct {
+	buildOptions common.BuildFlagsWrapper
+	local        bool
+	platforms    []string
+}
+
+var (
+	farmBuildDescription = `Builds a container image using podman system connections, then bundles them into a manifest list.`
+	buildCommand         = &cobra.Command{
+		Use:     "build",
+		Short:   "Build a container image for multiple architectures",
+		Long:    farmBuildDescription,
+		RunE:    build,
+		Example: "podman farm build [flags] buildContextDirectory",
+		Args:    cobra.ExactArgs(1),
+	}
+	buildOpts = buildOptions{
+		buildOptions: common.BuildFlagsWrapper{},
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: buildCommand,
+		Parent:  farmCmd,
+	})
+	flags := buildCommand.Flags()
+	flags.SetNormalizeFunc(utils.AliasFlags)
+
+	localFlagName := "local"
+	// Default for local is true and hide this flag for the remote use case
+	if !registry.IsRemote() {
+		flags.BoolVarP(&buildOpts.local, localFlagName, "l", true, "Build image on local machine as well as on farm nodes")
+	}
+	cleanupFlag := "cleanup"
+	flags.BoolVar(&buildOpts.buildOptions.Cleanup, cleanupFlag, false, "Remove built images from farm nodes on success")
```

Comment:
Shouldn't this be `true` by default ? Or is it `false` with intention keep `cache` for future builds ?

Reply:
yup, idea is to leave cache around by default as we do with `podman build`. The user can enable this if they would like to cleanup after builds.